File: app/src/main/java/com/bas/google_book_app/ui/login/LoginActivity.java
package com.bas.google_book_app.ui.login;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.bas.google_book_app.MainActivity;
import com.bas.google_book_app.R;
import com.bas.google_book_app.databinding.ActivityLoginBinding;
import com.bas.google_book_app.ui.register.RegisterActivity;
import com.bas.google_book_app.utilsdata.UserSession;

import java.util.HashMap;

public class LoginActivity extends AppCompatActivity {

    private ActivityLoginBinding mBinding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_login);

        mBinding.tvRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
            }
        });

        mBinding.tbnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                setLogin();
            }
        });

    }

    private void setLogin() {
        if (mBinding.edtUsername.getText().toString().isEmpty()){
            Toast.makeText(this, "Username can't Empty", Toast.LENGTH_SHORT).show();
        }else   if (mBinding.edtPass.getText().toString().isEmpty()){
            Toast.makeText(this, "Password can't Empty", Toast.LENGTH_SHORT).show();
        }else {
            HashMap<String, String> getUserDetails =  new UserSession(this).getUserDetails();
            String username = getUserDetails.get("Name");
            String password = getUserDetails.get("txtPassword");
            if (mBinding.edtUsername.getText().toString().equals(username) &&
                    mBinding.edtPass.getText().toString().equals(password)){
                Toast.makeText(this, "Login Success", Toast.LENGTH_SHORT).show();
                startActivity(new Intent(LoginActivity.this, MainActivity.class));
                new UserSession(this).isUserLoggedIn();

            }else {
                Toast.makeText(this, "Username or Password is wrong", Toast.LENGTH_SHORT).show();

            }

        }
    }
}

File: app/src/main/java/com/bas/google_book_app/ui/login/LoginActivity.kt
package com.bas.google_book_app.ui.login

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.bas.google_book_app.MainActivity
import com.bas.google_book_app.R
import com.bas.google_book_app.databinding.ActivityLoginBinding
import com.bas.google_book_app.ui.login.LoginActivity
import com.bas.google_book_app.ui.register.RegisterActivity
import com.bas.google_book_app.utilsdata.UserSession

class LoginActivity : AppCompatActivity() {
    private var mBinding: ActivityLoginBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        mBinding?.tvRegister?.setOnClickListener {
            startActivity(
                Intent(
                    this@LoginActivity,
                    RegisterActivity::class.java
                )
            )
        }
        mBinding?.tbnLogin?.setOnClickListener { setLogin() }
    }

    private fun setLogin() {
        if (mBinding?.edtUsername?.text.toString().isEmpty()) {
            Toast.makeText(this, "Username can't Empty", Toast.LENGTH_SHORT).show()
        } else if (mBinding?.edtPass?.text.toString().isEmpty()) {
            Toast.makeText(this, "Password can't Empty", Toast.LENGTH_SHORT).show()
        } else {
            val getUserDetails = UserSession(this)?.getUserDetails()
            val username = getUserDetails?.get("Name")
            val password = getUserDetails?.get("txtPassword")
            if (mBinding?.edtUsername?.text.toString() == username &&
                mBinding?.edtPass?.text.toString() == password) {
                Toast.makeText(this, "Login Success", Toast.LENGTH_SHORT).show()
                startActivity(Intent(this@LoginActivity, MainActivity::class.java))
                UserSession(this)?.isUserLoggedIn()
            } else {
                Toast.makeText(this, "Username or Password is wrong", Toast.LENGTH_SHORT).show()
            }
        }
    }
}